subiquity/server/apt.py

```diff
@@ -99,6 +130,67 @@ async def apply_apt_config(self, context):
             self.app, context, 'apt-config', '-t', self.configured_tree.p(),
             config=config_location, private_mounts=True)
 
+    async def run_apt_config_check(self, output: io.StringIO) -> None:
+        """ Run apt-get update (with various options limiting the amount of
+        data donwloaded) in the overlay where the apt configuration was
+        previously deployed. The output of apt-get (stdout + stderr) will be
+        written to the output parameter.
+        Raises a AptConfigCheckError exception if the apt-get command exited
+        with non-zero. """
+        assert self.configured_tree is not None
+
+        pfx = pathlib.Path(self.configured_tree.p())
+
+        apt_dirs = {
+            "Etc::SourceList": pfx / "etc/apt/sources.list",
+            "Etc::SourceParts": pfx / "etc/apt/sources.list.d",
+            "Etc::Main": pfx / "etc/apt/apt.conf",
+            "Etc::Parts": pfx / "etc/apt/apt.conf.d",
+            "Etc::Preferences": pfx / "etc/apt/preferences",
+            "Etc::PreferencesParts": pfx / "etc/apt/preferences.d",
+            "Cache::Archives": pfx / "var/lib/apt/archives",
+            "State::Lists": pfx / "var/lib/apt/lists",
+            "Cache::PkgCache": None,
+            "Cache::SrcPkgCache": None,
+        }
+
+        # Need to ensure the "partial" directory exists.
+        partial_dir = apt_dirs["State::Lists"] / "partial"
+        partial_dir.mkdir(
+                parents=True, exist_ok=True)
+        try:
+            shutil.chown(partial_dir, user="_apt")
+        except (PermissionError, LookupError) as exc:
+            log.warning("could to set owner of file %s: %r", partial_dir, exc)
```

Comment:
This call can run in dry-run but it will fail (unless run as root).
.. which is okay since apt will only try to `setuid("_apt")` if it is running as root.

OTOH, when doing a real install:
 * the chown call succeeds (or should succeed)
 * apt still fails to use the sandbox _apt user because the overlay is not accessible. so it warns and runs unsandboxed. I would like to get this fixed.